<!-- -*- mode: XML -*- -->
<launch>
    <arg name="racecar_version" default="racecar-v2" />
    <arg name="run_camera" default="false"/>
    <arg name="map_dir" value="$(find rl_node)/../../../maps/"/>
    <arg name="map" value="$(arg map_dir)/inf_20220212.yaml"/>

    <include file="$(find racecar)/launch/includes/$(arg racecar_version)-teleop.launch.xml">
        <arg name="racecar_version" value="$(arg racecar_version)" />
        <arg name="run_camera" value="$(arg run_camera)" />
    </include>

    <include file="$(find phidgets_spatial)/launch/spatial.launch"></include>

    <!-- Launch safety node -->
   <!-- <node pkg="rl_node" type="safety_node.py" name="safety_node" output="screen"/>
-->
    <!-- Launch velocity filter -->
    <node pkg="rl_node" type="filter_node.py" name="filter_node" output="screen">
        <rosparam command="load" file="$(find rl_node)/cfg/filter_node.yaml"/>
    </node>

    <!-- Launch controller node -->
    <remap from="drive" to="vesc/high_level/ackermann_cmd_mux/input/nav_0" />
    <include file="$(find rl_node)/launch/only_agent.launch">
        <arg name="params" value="params_hw.yaml"></arg>
    </include>
</launch>
